`IconTile` documentation update (#2514)
hashicorp/design-system: website/docs/components/icon-tile/partials/guidelines/guidelines.md

## Color

Use **Neutral** if the object or page is not a specific product feature but something universal. For example, for a “Dashboard” or “User” page.

![Icon Tile with neutral color and dashboard icon](/assets/components/icon-tile/icon-tile.png)

Use a **product-specific color** for objects or pages directly related to a product. For example:

- For a page showing a “Consul cluster”
- In a card or table row that represents a “Consul cluster”

![Product specific Icon Tiles](/assets/components/icon-tile/icon-tiles-colors.png)

<Doc::ImageCaption @text="Examples of product-specific Icon Tiles"/>

## Size

Medium is the default size, but we recommend using the size that best fits the supporting text or UI. For example, don’t use large Icon Tiles in tables.

![Icon Tile sizes](/assets/components/icon-tile/icon-tiles-sizes.png)

## Secondary icon

A secondary icon can be added to provide additional context. For example, for an “Add user” page the “plus” icon indicates the action.

![Icon Tiles with secondary icon](/assets/components/icon-tile/icon-tiles-scondary-icon.png)

## Using with content

> Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent as decorative element or accessory.

Reviewer: ```suggestion
Use Icon Tiles as a decorative element or accessory next to a text label for the object or page they are being used to represent. Icon Tiles should not be used without accompanying content.
```

Reviewer: The current sentence is difficult to parse. I attempted to simplify and clarify in the above suggestion but feel free to tweak a bit more...

### When not to use

> - Within a paragraph where an object is referred to.

Reviewer 2: I'm not really sure what you mean by this, can you clarify a bit?

Author: I can remove this snippet of text. I think what I was trying to say is, "Don't use outside of headings and standalone visuals." in a vague way 😅

## Color

Use **Neutral** if the object or page is not a specific product feature but something universal. For example, for a “Dashboard” or “User” page.

![Icon Tile with neutral color and dashboard icon](/assets/components/icon-tile/icon-tile.png)

Use a **product-specific color** for objects or pages directly related to a product. For example:

- For a page showing a “Consul cluster”
- In a card or table row that represents a “Consul cluster”

![Product specific Icon Tiles](/assets/components/icon-tile/icon-tiles-colors.png)

<Doc::ImageCaption @text="Examples of product-specific Icon Tiles"/>

## Size

Medium is the default size, but we recommend using the size that best fits the supporting text or UI. For example, don’t use large Icon Tiles in tables.

![Icon Tile sizes](/assets/components/icon-tile/icon-tiles-sizes.png)

## Secondary icon

A secondary icon can be added to provide additional context. For example, for an “Add user” page the “plus” icon indicates the action.

![Icon Tiles with secondary icon](/assets/components/icon-tile/icon-tiles-scondary-icon.png)

## Using with content

> Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.

Reviewer 2: [suggestion] not opinionated on this, but this could be a way to combine the final sentence with the rest of the content, it seems like it's a bit of a fragment currently.

```suggestion
Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent as decorative element or accessory.
```

## Using with content

Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.

> ## Near interactive elements

Reviewer 3: To follow the standard from the previous heading, maybe rename this to:

```suggestion
## Using with interactive elements
```

> Avoid placing Icon Tiles next to buttons or other clickable elements. This can confuse users and make it unclear what is interactive. In these cases, use a simple icon for better usability.

Reviewer 3: I tried rewording this so that it can be a little more descriptive (plus providing a source) as to why we want folks to be mindful of this.

```suggestion
Avoid placing Icon Tiles next to visually similar interactive elements. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable) is an accessibility standard that deters interfaces from using elements that look the same but behave differently in close visual proximity. This can cause user confusion through the established visual language. 
```

> !!! Dont
> 
> ![Dont example of Icon Tile placed in the same proximity of other interactive elements](/assets/components/icon-tile/icon-tile-example.png)
> !!!
> 
> !!! Do
> 
> ![Do example of icon placed in the same proximity of other interactive elements, instead of an Icon Tile](/assets/components/icon-tile/icon-tile-example-do.png)
> !!!

Author: ```suggestion

```

## Using with content

Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.

## Using with interactive elements

> Avoid placing Icon Tiles adjacent to visually similar interactive elements, as this can lead to user confusion due to lack of visual differentiation. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity. 

Reviewer 3: We should reword this a bit to say something along the lines of:

```suggestion
Avoid placing Icon Tiles immediately next to secondary icon only button variants. Even though they visually look different, their proximity can confuse users because they share similar characteristics. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity. 
```

Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.

## Using with interactive elements

> Avoid placing Icon Tiles next to visually similar interactive elements. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable) is an accessibility standard that deters interfaces from using elements that look the same but behave differently in close visual proximity. This can cause user confusion through the established visual language. 

Reviewer 2: I might suggest merging the last sentence into one of the previous points, otherwise it seems a bit like a fragment. I'm also not sure about "established visual language", that seems like an ambiguous statement. What about this:

```suggestion
Avoid placing Icon Tiles adjacent to visually similar interactive elements, as this can lead to user confusion due to lack of visual differentiation. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity. 
```

However, it bears mentioning that if we're recommending consumers not place Icon Tiles next to interactive elements, is the visual language of the Icon Tile to blame? does it appear _too_ interactive?

Reviewer: I think it would be worthwhile to explore a design update to the IconTile to make it less "interactive" looking. (Perhaps trying a solid color background without a border to differentiate more from Buttons?)

This is something that could be considered as a follow on though vs as part of this current PR/ticket.

Author: Yes, I agree. IconTile's visual design is worth discussing, especially considering Kristin's feedback. We should explore potential visual changes to make it appear less interactive.

Author: > Not necessarily a problem, just noting the overlap, or maybe there's another example that could be used?
I'll spin up another example, but I feel the one Melanie supplied in the Jira ticket makes the most sense as an example since I haven't seen many others where an IconTile is in that close proximity to our second button.

Reviewer 3: FYI, played around with approach: https://www.figma.com/design/3VZiWIbcxIahVfVohQzhPk/Majed's-Playground?node-id=478-1032&t=4ZgbQi4ElepOVdVk-1

Ultimately the problem is that we use the same visual language as our buttons (border and shadow) which makes it similar. If we strip them, it'll look less interactive.

Thoughts @andgen404 ?

Reviewer: Just a mention from the meeting, I'd also try using a solid color vs. a gradient to see how that works.
